internal/secretsource/url/url.go

```diff
@@ -0,0 +1,256 @@
+)
+
+var (
+	errInvalidConfig                 = errors.New("config parameter is required in format 'config=path/to/config'")
+	errMissingURLTemplate            = errors.New("urlTemplate is required in config file")
+	errFailedToGetSecret             = errors.New("failed to get secret")
+	errInvalidRequestsPerMinuteLimit = errors.New("requestsPerMinuteLimit must be greater than 0")
+	errInvalidRequestsBurst          = errors.New("requestsBurst must be greater than 0")
+)
+
+// extConfig holds the configuration for URL-based secrets.
+type extConfig struct {
+	// URLTemplate is a URL template with {key} placeholder
+	// Example: "https://api.example.com/secrets/{key}"
+	URLTemplate string `json:"urlTemplate"`
+
+	// Headers to include in the request (e.g., Authorization headers)
+	Headers map[string]string `json:"headers"`
+
+	// Method is the HTTP method to use (defaults to GET)
+	Method string `json:"method"`
+
+	// ResponsePath is a JSON path to extract the secret value from the response
+	// Use dot notation for nested fields (e.g., "data.value")
+	// If empty, the entire response body is treated as the secret
+	ResponsePath string `json:"responsePath"`
+
+	// RequestsPerMinuteLimit sets the maximum requests per minute
+	RequestsPerMinuteLimit *int `json:"requestsPerMinuteLimit"`
+
+	// RequestsBurst allows a burst of requests above the rate limit
+	RequestsBurst *int `json:"requestsBurst"`
+
+	// Timeout for HTTP requests in seconds (defaults to 30)
+	TimeoutSeconds *int `json:"timeoutSeconds"`
+}
+
+const (
+	// The rate limiter replenishes tokens in the bucket once every 200 ms
+	// (5 per second) and allows a burst of 10 requests firing faster than
+	// that. If the client keeps making requests at the rapid pace, they
+	// will be slowed down. This allows a client to ask for a bunch of
+	// secrets at the start of a script, and then it slows it down to a
+	// reasonable pace.
+	defaultRequestsPerMinuteLimit = 300 // 300 requests per minute is one request every 200 ms
+	defaultRequestsBurst          = 10  // Allow a burst of 10 requests
+	defaultTimeoutSeconds         = 30  // 30 seconds timeout
+)
+
+//nolint:gochecknoinits // This is how k6 secret source registration works.
+func init() {
+	secretsource.RegisterExtension("url", func(params secretsource.Params) (secretsource.Source, error) {
+		config, err := getConfig(params.ConfigArgument, params.FS)
+		if err != nil {
+			return nil, fmt.Errorf("missing or invalid config: %w", err)
+		}
+
+		timeout := time.Duration(*config.TimeoutSeconds) * time.Second
+
+		return &urlSecrets{
+			config: config,
+			httpClient: &http.Client{
+				Timeout: timeout,
+			},
+			limiter: newLimiter(*config.RequestsPerMinuteLimit, *config.RequestsBurst),
+		}, nil
+	})
+}
+
+type urlSecrets struct {
+	config     extConfig
+	httpClient *http.Client
+	limiter    limiter
+}
+
+func (us *urlSecrets) Description() string {
+	return fmt.Sprintf("URL-based secret source from %s", us.config.URLTemplate)
```

Comment:
If this gets logged, is there any chance that some sensitive information leaks by this mean?